e2e/transaction_test.go

```diff
@@ -1,76 +1,85 @@
 package e2e
 
 import (
-	"fmt"
+	"context"
 	"math/big"
+	"os"
 	"testing"
 	"time"
 
-	"github.com/stretchr/testify/assert"
-	"github.com/umbracle/go-web3"
-	"github.com/umbracle/go-web3/wallet"
-
+	"github.com/0xPolygon/minimal/crypto"
 	"github.com/0xPolygon/minimal/e2e/framework"
-	"github.com/0xPolygon/minimal/helper/hex"
 	"github.com/0xPolygon/minimal/types"
-)
-
-var (
-	addr0 = types.Address{}
-	addr1 = types.Address{0x1}
+	"github.com/stretchr/testify/assert"
+	"github.com/umbracle/go-web3"
 )
 
 func TestSignedTransaction(t *testing.T) {
-	fr := framework.NewTestServerFromGenesis(t)
+	signer := &crypto.FrontierSigner{}
+	senderKey, senderAddr := framework.GenerateKeyAndAddr(t)
+	_, receiverAddr := framework.GenerateKeyAndAddr(t)
 
-	// 0xdf7fd4830f4cc1440b469615e9996e9fde92608f
-	var privKeyRaw = "0x4b2216c76f1b4c60c44d41986863e7337bc1a317d6a9366adfd8966fe2ac05f6"
-	key, err := wallet.NewWalletFromPrivKey(hex.MustDecodeHex(privKeyRaw))
-	assert.NoError(t, err)
+	dataDir, err := framework.TempDir()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	preminedAmount := framework.EthToWei(10)
+	ibftManager := framework.NewIBFTServersManager(t, IBFTMinNodes, dataDir, IBFTDirPrefix, func(i int, config *framework.TestServerConfig) {
+		config.Premine(types.Address(senderAddr), preminedAmount)
```

Comment:
Redundant type conversion
```suggestion
		config.Premine(senderAddr, preminedAmount)
```